File: t/PACSFramework/CS/MedicalWorkstationCustomizationDemo/UI/CustomizedMenuStrip/CustomizedMenuStrip.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using Leadtools.Medical.Workstation.Interfaces;
using Leadtools.Medical.Workstation.UI;

namespace Leadtools.Demos.Workstation.Customized
{
   class ExamplesMenuStrip : MenuStrip
   {
      #region Public
         
         #region Methods
         
            public ExamplesMenuStrip ( ) 
            {
            }
            
            public void RegisterFeatures ( IWorkstationStripItemFeatureExecuter toolStripExecuter )
            {
               if ( null == toolStripExecuter ) 
               {
                  throw new ArgumentNullException ( ) ;
               }
               
               if ( ToolStripController != null ) 
               {
                  UnregisterFeatures ( ) ;
               }
               
               ToolStripController = toolStripExecuter ;
               
               ToolStripController.RegisterToolStripMenuItemFeatures ( this ) ;
            }
            
            public void UnregisterFeatures ( )
            {
               if ( null == ToolStripController )
               {
                  return ;
               }
               
               ToolStripController.UnregisterToolStripMenuItemFeatures ( this ) ;
            }
         
         #endregion
         
         #region Properties
         
            public IWorkstationStripItemFeatureExecuter ToolStripController
            {
               get ;
               private set ;
            }
            
         #endregion
         
         #region Events
            
         #endregion
         
         #region Data Types Definition
            
         #endregion
         
         #region Callbacks
            
         #endregion
         
      #endregion
      
      #region Protected
         
         #region Methods
            
         #endregion
         
         #region Properties
            
         #endregion
         
         #region Events
            
         #endregion
         
         #region Data Members
            
         #endregion
         
         #region Data Types Definition
            
         #endregion
         
      #endregion
      
      #region Private
         
         #region Methods
            
         #endregion
         
         #region Properties
            
         #endregion
         
         #region Events
            
         #endregion
         
         #region Data Members
            
         #endregion
         
         #region Data Types Definition
            
         #endregion
         
      #endregion
      
      #region internal
         
         #region Methods
            
         #endregion
         
         #region Properties
            
         #endregion
         
         #region Events
            
         #endregion
         
         #region Data Types Definition
            
         #endregion
         
         #region Callbacks
            
         #endregion
         
      #endregion
   }
}

File: t/PACSFramework/VB/MedicalWorkstationCustomizationDemo/UI/CustomizedMenuStrip/CustomizedMenuStrip.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports Leadtools.Medical.Workstation.Interfaces
Imports Leadtools.Medical.Workstation.UI

Namespace Leadtools.Demos.Workstation.Customized
   Class ExamplesMenuStrip
      Inherits MenuStrip
#Region "Public"

#Region "Methods"

      Public Sub New()
      End Sub
      Public Sub RegisterFeatures(ByVal toolStripExecuter As IWorkstationStripItemFeatureExecuter)
         If Nothing Is toolStripExecuter Then
            Throw New System.ArgumentNullException()
         End If

         If Not ToolStripController Is Nothing Then
            UnregisterFeatures()
         End If

         ToolStripController = toolStripExecuter

         ToolStripController.RegisterToolStripMenuItemFeatures(Me)
      End Sub

      Public Sub UnregisterFeatures()
         If ToolStripController Is Nothing Then
            Return
         End If

         ToolStripController.UnregisterToolStripMenuItemFeatures(Me)
      End Sub

#End Region

#Region "Properties"

      Public ToolStripController As IWorkstationStripItemFeatureExecuter

#End Region

#Region "Events"

#End Region

#Region "Data Types Definition"

#End Region

#Region "Callbacks"

#End Region

#End Region

#Region "Protected"

#Region "Methods"

#End Region

#Region "Properties"

#End Region

#Region "Events"

#End Region

#Region "Data Members"

#End Region

#Region "Data Types Definition"

#End Region

#End Region

#Region "Private"

#Region "Methods"

#End Region

#Region "Properties"

#End Region

#Region "Events"

#End Region

#Region "Data Members"

#End Region

#Region "Data Types Definition"

#End Region

#End Region

#Region "internal"

#Region "Methods"

#End Region

#Region "Properties"

#End Region

#Region "Events"

#End Region

#Region "Data Types Definition"

#End Region

#Region "Callbacks"

#End Region

#End Region
   End Class
End Namespace
